taskvine/src/worker/vine_worker.c

```diff
@@ -520,14 +541,25 @@ static int start_process(struct vine_process *p, struct link *manager)
 		vine_gpus_allocate(t->resources_requested->gpus, t->task_id);
 	}
 
+	/* Starting a function call is different from starting a standard task, so we return early. */
+	if (t->needs_library) {
+		itable_insert(procs_running, p->task->task_id, p);
+		debug(D_VINE, "started function call %d: %s", p->task->task_id, p->task->command_line);
+		p->library_process->functions_running++;
+		return start_function(p);
```

Comment:
I would prefer that all task types flow through the function to the end, so that they get treated mostly the same.

For example, this code here skips the important step at the bottom of inserting `p` into `procs_running`.

Reply:
I inserted the process to procs_running in the beginning, but this makes sense. Fixed!